modules/@apostrophecms/schema/ui/apos/lib/detectChange.js

```diff
@@ -0,0 +1,23 @@
+// Detect whether a document or document field has been changed,
+// in a way that warrants confirmation before exiting a modal,
+// or emitting an update event
+
+import { isEqual } from 'lodash';
+
+export function detectDocChange(schema, v1, v2) {
+  return schema.some(field => {
+    return detectFieldChange(field, v1[field.name], v2[field.name]);
+  });
+}
+
+export function detectFieldChange(field, v1, v2) {
```

Comment:
Right now we don't look at field here but this is very much subject to change if we find this simple algorithm in adequate to the needs of some field types. Futureproofing.